### PR TITLE
Env var exit code

### PR DESCRIPTION
**Pull request by dhuckins**

### CODE REVIEW

File: command/agent/exec/server.go [hunk 1]
```diff
@@ -153,8 +156,7 @@ func (s *Server) Run(ctx context.Context, envTmpls map[string]*config.EnvTemplat
 			s.runner.Stop()
 			s.childLock.Lock()
 			if s.child != nil {
-				// TODO: use kill to immediately kill or gracefully?
-				s.child.Kill()
+				s.child.Stop()
```
Comment (averche) [inline command/agent/exec/server.go:159]: Does this hard-kill the process after a timeout or do we need to do something special on top?

File: command/agent/exec/server.go [hunk 2]
```diff
@@ -205,6 +207,9 @@ func (s *Server) Run(ctx context.Context, envTmpls map[string]*config.EnvTemplat
 					return fmt.Errorf("unable to bounce command: %w", err)
 				}
 			}
+		case exitCode := <-s.exitCh:
+			// process exited on its own
+			return &ProcessExitError{ExitCode: exitCode}
```
Comment (averche) [inline command/agent/exec/server.go:212]: 👍

**Review by averche — APPROVED** [review]
I like this approach!

File: command/agent/exec/server.go [hunk 3]
```diff
@@ -60,6 +60,14 @@ type Server struct {
 	exitCh chan int
 }
 
+type ProcessExitError struct {
```
Comment (averche) [inline command/agent/exec/server.go:63]:
```suggestion
type processExitError struct {
```
Might want to make it unexported

File: command/agent/exec/server.go (hunk 1 above)
Comment (dhuckins) [reply command/agent/exec/server.go:159]: https://pkg.go.dev/github.com/hashicorp/consul-template/child#Child.Stop
> Stop behaves almost identical to Kill except it suppresses future processes from being started by this child and it prevents the killing of the child process from sending its value back up the exit channel. This is useful when doing a graceful shutdown of an application.

_think_ we're good?

**Pull request merged**